### PR TITLE
Improve ICA docstring, the second

### PR DESCRIPTION
**Pull request by jeythekey**
One more attempt to make the docstring of ICA more explicit, especially as to the three different `*_components_*` parameters.

Also fixing wrong shape specifications of the attributes `pca_components_` and `pca_explained_variance_`.

#### Reference issue
Expanding on #5114, in reference to #4856.

#### What does this implement/fix?
V.s.; in detail:
* Adding a (first attempt of an) introduction about pre-ICA PCA, also referencing the warning about dimensionality reduction mentioned e.g. [here](https://github.com/mne-tools/mne-python/issues/4856#issuecomment-377194869).
!! The formulation about the whitening needs to be reviewed, I am not sure, if this is correct! See below.
* Referencing the most important methods which make use of the various attributes and parameters, especially thereby trying to be more explicit in which steps the various component-parameters are used.
* Some re-formulations.
* Mentioning Evoked as supported container.

#### OPEN QUESTIONS:

1. `max_pca_components`:
I am not sure if (with now explicitly specifying the sklearn.PCA svd_solver to 'full' as of sklearn 0.18.) max_pca_components has any influence on the actual PCA (e.g. by influencing the choice of the algorithm used; I am referring to this of the PCA docstring: "If ``n_components == 'mle'`` and ``svd_solver == 'full'``, Minka's MLE is used to guess the dimension.") 
If this would _not_ be the case, the only other use case I could think of for including 2 separate n-PCA-parameters for decomp and re-comp is that the user makes up his mind later, after fitting, about how many PCA-compos to include in the reconstruction. But would it then hurt (data size etc.) to always return and store simply all PCA-compos?
If the latter would also _not_ be the case, I think this parameter is completely redundant and should be merged with `n_pca_components` (which has to be >= `n_components_`), as it is probably the primary source of confusion to have two parameters related to the number of PCA components.
--> Could someone check this?

2. I am not sure about the whitening/sphering:
In `_fit()`, the variance-normalization is somehow applied to both the pca_components_ and the (un)mixing_matrix_ (once * and once /, though):
```python
    593: self.pca_components_ *= np.sqrt(exp_var[:, None])  # (or in pca.fit_transform())
    ...
    619: self.unmixing_matrix_ /= np.sqrt(exp_var[sel])[None, :]  # whitening
    620: self.mixing_matrix_ = linalg.pinv(self.unmixing_matrix_)
```
Then, later in _pick_sources, the two are combined:
```python
    unmixing = np.eye(_n_pca_comp)
    unmixing[:n_components, :n_components] = self.unmixing_matrix_
    unmixing = np.dot(unmixing, self.pca_components_[:_n_pca_comp])
```
So, I wonder, if it is correct to say that the `(un)mixing_` matrix is whitened? Or would you rather way it is "also whitening the data"?
And could I think of applying the normalization in this way is in the end cancelling itself out when pca-unmixing and and ica-unmixing are being combined (i.e. in the complete unmixing)?

This is actually my first commit/PR ever, so please bear with me ;)

### CODE REVIEW

**Comment by jeythekey** [comment]
opened here by mistake, the real one is [here](https://github.com/mne-tools/mne-python/pull/5620) (and was merged).
